packages/swingset-liveslots/src/virtualReferences.js

```diff
@@ -39,9 +39,19 @@ export function makeVirtualReferenceManager(
   const droppedCollectionRegistry = new FinalizationRegistry(
     finalizeDroppedCollection,
   );
+  // Our JS engine is configured to treat WeakRefs as strong during
+  // organic (non-forced GC), to minimize execution variation. To
+  // prevent FinalizationRegistry callbacks from varying this way, we
+  // must maintain WeakRefs to everything therein. This Map maps from
+  // "holder" objects to a WeakRef around the real target, and the map
+  // entry is deleted when the FR fires.
+  const droppedCollectionWeakRefs = new Map(); // holder -> WeakRef(target)
 
   function registerDroppedCollection(target, descriptor) {
-    droppedCollectionRegistry.register(target, descriptor);
+    const holder = {};
+    const wr = new WeakRef(target);
+    droppedCollectionWeakRefs.set(holder, wr);
+    droppedCollectionRegistry.register(target, { descriptor, holder });
```

Comment:
We don't need the droppedCollectionWeakRefs indirection, the heldValue is held strongly.

```suggestion
    const wr = new WeakRef(target);
    droppedCollectionRegistry.register(target, { descriptor, wr });
```

Reply:
Ok, will do, but I think that will change my testing plan.

I can check that a mock WeakRef was created around the VOAware collection. I can instrument my mock FR to let me inspect the `heldValue` and confirm that it is present in there. But I won't be able to test past that point. I guess that's probably good enough.